### ISSUE TITLE
CORS Problem CalDav

### ISSUE THREAD

**Issue by guitar9** [nextcloud/server#19719]
Hi I am using nextcloud tasks and I want to build a client to get the task Informations.
This is the request  I am doing:

```
    const response = await fetch("URL/remote.php/dav/", {
      method: 'PROPFIND', // *GET, POST, PUT, DELETE, etc.
      mode: 'cors', // no-cors, *cors, same-origin
      headers: {
        'Content-Type': 'application/xml',
        'Authorization': 'Basic ' + encode("user" + ":" + "password"),
        // 'Content-Type': 'application/x-www-form-urlencoded',
      },
      body: `<x0:propfind xmlns:x0="DAV:"><x0:prop><x0:current-user-principal/></x0:prop></x0:propfind>`// body data type must match "Content-Type" header
    });
    const text =  await response.text(); // parses JSON response into native JavaScript objects
    console.log(text)
```
 I get this error in the browser:
` Access to fetch at 'URL/remote.php/dav/' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by kesselb** [comment]
Duplicate of https://github.com/nextcloud/server/issues/3131

Nextcloud does not respond to preflight request correctly. A common workaround is to proxy the request locally.

**Comment by guitar9** [comment]
How does then all apps work with CalDav Sync with Nextcloud Calender?
When I only can make request from same domain...

**Comment by kesselb** [comment]
They ignore the CORS header.

**Comment by guitar9** [comment]
Ah ok thank you very much.
So I can for example disable cors with react native or flutter?
